src/main/java/stsjorbsmod/patches/BottledMemoryPatch.java

```diff
@@ -0,0 +1,113 @@
+            if (methodCall.getClassName().equals(CardGroup.class.getName()) && methodCall.getMethodName().equals(methodName)) {
+                methodCall.replace("{ $_ = (stsjorbsmod.patches.BottledMemoryPatch.removeBottledCards($proceed())); }");
+            }
+        }
+    }
+
+    /**
+     * Removes all bottled cards from the card group passed in and returns the CardGroup.
+     *
+     * @param cards
+     * @return cards
+     */
+    public static CardGroup removeBottledCards(CardGroup cards) {
+        cards.group.removeIf(c -> c.inBottleTornado || c.inBottleFlame || c.inBottleLightning || AbstractCardMemoryFields.inBottleMemory.get(c));
+        return cards;
+    }
+
+    @SpirePatch(clz = BottledTornado.class, method = "onEquip")
+    public static class BottledTornado_onEquip {
+        public static ExprEditor Instrument() {
+            return new BottledGetCardsExprEditor("getPowers");
+        }
+    }
+
+    @SpirePatch(clz = BottledFlame.class, method = "onEquip")
+    public static class BottledFlame_onEquip {
+        public static ExprEditor Instrument() {
+            return new BottledGetCardsExprEditor("getAttacks");
+        }
+    }
+
+    @SpirePatch(clz = BottledLightning.class, method = "onEquip")
+    public static class BottledLightning_onEquip {
+        public static ExprEditor Instrument() {
+            return new BottledGetCardsExprEditor("getAttacks");
+        }
+    }
+
+    @SpirePatch(clz = AbstractPlayer.class, method = "bottledCardUpgradeCheck", paramtypez = {AbstractCard.class})
+    public static class AbstractPlayer_bottledCardUpgradeCheck {
+        @SpirePostfixPatch
+        public static void patch(AbstractPlayer __this, AbstractCard c) {
+            if (AbstractCardMemoryFields.inBottleMemory.get(c) && __this.hasRelic(BottledMemoryRelic.ID)) {
+                ((BottledMemoryRelic) __this.getRelic(BottledMemoryRelic.ID)).setDescriptionAfterLoading();
+            }
+        }
+    }
+
+    public static boolean isInBottleMemory(AbstractCard c) {
+        return AbstractCardMemoryFields.inBottleMemory.get(c);
+    }
+
+    @SpirePatch(clz = CardGroup.class, method = "initializeDeck", paramtypez = {CardGroup.class})
+    public static class AbstractGroup_initializeDeck {
+        public static ExprEditor Instrument() {
+            return new ExprEditor() {
+                @Override
+                public void edit(FieldAccess fieldAccess) throws CannotCompileException {
+                    // Transforms `!inBottleTornado` into `!(inBottleTornado || inBottleMemory)`
+                    // Logically equivalent to `!inBottleTornado && !inBottleMemory`
+                    if (fieldAccess.getClassName().equals(AbstractCard.class.getName()) && fieldAccess.getFieldName().equals("inBottleTornado")) {
+                        fieldAccess.replace("{ $_ = ($proceed() || stsjorbsmod.patches.BottledMemoryPatch.isInBottleMemory(c)); }");
```

Comment:
I preferred following pre-existing bottled relic draw patterns rather the example in the DefaultModBase